### ISSUE TITLE
Unlimited TK dodge after window switching

### ISSUE THREAD

**Issue by hikawasayotohina** [ArranzCNL/ImprovedCameraSE#105]
This is rare, and usually occurs when switching back to the game window after switching from the game window to another application's window. Of course, it may also be related to the player setting the key on the TK Dodge to Alt, because the key on the window switching is Alt+Tab, which may be related in some way?

"But it may be difficult to convince, because it is difficult to imagine this situation being caused by this Mod."

And I'm not sure if other players have the same problem because it rarely happens and it's not easy to witness and record videos

However, I have noticed that when the bug occurs, the player cannot use Tab to open the menu until the player presses Alt again, the dodge action stops, and the Tab menu can be opened normally

That's why I said it might be related to Windows window switching

but

Does this sound like a bug for TK Dodge?

It's just that this has never happened in previous versions

Regardless of whether or not only TK Dodge RE is installed, it will not happen

The latest version of ImprovedCamerasE PR5 is installed

And also install TK Dodge RE, which may occur

Because it rarely happens, and there's no video evidence

Once I get the chance, I'll record it

I love switching to other windows to do other things when the game loads

After the game is loaded, I will switch back to the game

At this point, the bug may occur

"This is so touchless that I have encountered it many times and cannot record it."

**Comment by hikawasayotohina** [comment]
Oh! I recorded the moment the bug triggered

I am uploading a video

**Comment by valepu** [comment]
I don't know if you have tried already, but what happens if you press the ALT key again once you are back on Skyrim and the bug occurs?

**Comment by hikawasayotohina** [comment]
https://youtu.be/jI7Fl8aIBWs

**Comment by hikawasayotohina** [comment]
I recorded a scenario where this issue occurred. Please check

But I don't know if anyone else has this problem

Could this be an example?

But I really only have this happen if I install Improved Camera SE.

**Comment by hikawasayotohina** [comment]
> I don't know if you have tried already, but what happens if you press the ALT key again once you are back on Skyrim and the bug occurs?
Pressing Alt when the bug is triggered will stop the bug
I uploaded a video recording the occurrence of the bug until I press Alt to end the bug

**Comment by valepu** [comment]
Sorry I missed a piece on your first post where you mentioned something like that. It is a common occurrence though, that when you press a key and change window, some applications or games might act like the key is still pressed, and pressing it again fixes it.

So I'd like to ask one more question because I don't feel like it's clear from your ticket, does it also happen if you have TK Dodge installed but not ImprovedCamera?

**Comment by ArranzCNL** [comment]
TLDR it's technically a bethesda bug. What I did inside of Improved Camera is corrected the Skyrim window behaviour, as for some unknown reason bethesda decided to use DirectInput for keyboard/mouse controls.

It is just one of those things just remember to press the key again if alt+tabbing windows.

**Comment by hikawasayotohina** [comment]
> Sorry I missed a piece on your first post where you mentioned something like that. It is a common occurrence though, that when you press a key and change window, some applications or games might act like the key is still pressed, and pressing it again fixes it.
> 
> So I'd like to ask one more question because I don't feel like it's clear from your ticket, does it also happen if you have TK Dodge installed but not ImprovedCamera?
Sorry, it took me 5 days to reply
ImprovedCamera is not installed
When only TK Dodge is installed
This won't happen
